alyx/actions/admin.py

```diff
@@ -104,6 +106,53 @@ def queryset(self, request, queryset):
             return queryset.all()
 
 
+class HasNarrativeFilter(DefaultListFilter):
+    title = 'narrative'
+    parameter_name = 'has_narrative'
+
+    def lookups(self, request, model_admin):
+        return (
+            (None, 'All'),
+            ('narrative', 'Has narrative'),
+            ('no_narrative', 'No narrative'),
+        )
+
+    def queryset(self, request, queryset):
+        if self.value is not None:
+            regex_string = r'^(?: *|auto-generated session)$'
```

Comment:
What's the purpose of the group here? Also `\s` could be used to exclude all whitespace, not just spaces. FYI the regex here must be [PostgreSQL](https://www.postgresql.org/docs/current/functions-matching.html#POSIX-SYNTAX-DETAILS), not Python. The current expression you have here matches both.

Reply:
the non-matching group is necessary for the alternation expression (`|`) to work